File: stress/t/ctl_parity.f
      PROGRAM CTLPAR
      INTEGER*8 V,T,BITS,PAR,ODDP,I
      ODDP=0
      I=0
   10 IF (I.GE.64) GOTO 40
      V=I*2654435761_8
      V=IAND(V,1023_8)
      T=V
      BITS=0
   20 IF (T.EQ.0) GOTO 30
      IF (IAND(T,1_8).EQ.1) BITS=BITS+1
      T=ISHFT(T,-1)
      GOTO 20
   30 CONTINUE
      PAR=MOD(BITS,2_8)
      IF (PAR.EQ.1) ODDP=ODDP+1
      I=I+1
      GOTO 10
   40 CONTINUE
      WRITE(*,'(A,I0)') 'oddpar=',ODDP
      END

File: stress/t/ctl_parity.c
#include <stdio.h>
#include <stdint.h>
int main(void){
  int64_t v=0,t=0,bits=0,par=0,oddpar=0,i=0;
  for(i=0;i<64;i=i+1){
    v=(int64_t)((uint64_t)i*(uint64_t)2654435761ULL);
    v=v&1023;
    t=v;
    bits=0;
    while(t!=0){
      if((t&1)==1){ bits=bits+1; } else { bits=bits+0; }
      t=t>>1;
    }
    par=bits%2;
    if(par==1){ oddpar=oddpar+1; } else { oddpar=oddpar+0; }
  }
  printf("oddpar="); printf("%ld",(long)oddpar); printf("\n");
  return 0;
}
